internal/function/uast.go

```diff
@@ -197,11 +198,13 @@ func (f UAST) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 
 	resp, err := client.Parse(ctx, lang, bytes)
 	if err != nil {
-		return nil, ErrParseBlob.New(err)
+		logrus.Warn(ErrParseBlob.New(err))
```

Comment:
An error can be thrown an the protocol result be OK

Reply:
So, we should check if resp is nil too to fail here (I think).

Reply:
the protocol result shouldn't be `OK` if there is an error and if resp is not nil it's because the status if `FATAL`, so we must return any case

```
// DoWithContext does the same as Do(), but sopporting cancellation by the use
// of Go contexts.
func (r *ParseRequest) DoWithContext(ctx context.Context) (*protocol.ParseResponse, error) {
	if r.err != nil {
		return nil, r.err
	}

	resp, err := r.client.service.Parse(ctx, &r.internal)
	if err != nil {
		return nil, err
	} else if resp.Status == protocol.Fatal {
		return resp, FatalError(resp.Errors)
	}
	return resp, nil
}
```

Reply:
yep, you are right. But in the case that the error is not fatal, we should print the errors from the response as warnings and return the partial UAST.